### PR TITLE
Rewrite factorial as a product fix 8757

### LINKED ISSUE #8757: Rewrite factorial as a product
It would be nice that if, say,
```python
>>> k = Symbol('k', integer=True, nonnegative=True)
```
then the following would hold
```python
>>> factorial(k).rewrite(product)
Product(i, (i, 1, k))
```
(for a dummy positive integer `i`)

### PR DESCRIPTION
**Pull request by aktech**
fixes #8757 

``` python
>>> k = Symbol('k', integer=True, nonnegative=True)
>>> factorial(k).rewrite(Product)
Product(i, (i, 1, k)
```

### CODE REVIEW

**Comment by skirpichev** [comment]
Product, not product!

File: sympy/functions/combinatorial/factorials.py [hunk 1]
```diff
@@ -164,6 +164,11 @@ def eval(cls, n):
     def _eval_rewrite_as_gamma(self, n):
         return C.gamma(n + 1)
 
+    def _eval_rewrite_as_product(self, n):
```
Comment (skirpichev) [inline sympy/functions/combinatorial/factorials.py]: _Product

Comment (aktech) [inline sympy/functions/combinatorial/factorials.py]: Ah! Thanks.

File: sympy/functions/combinatorial/factorials.py [hunk 2]
```diff
@@ -164,6 +164,11 @@ def eval(cls, n):
     def _eval_rewrite_as_gamma(self, n):
         return C.gamma(n + 1)
 
+    def _eval_rewrite_as_Product(self, n):
+        if n.is_nonnegative and n.is_integer:
+            from sympy.abc import i
+            return C.Product(i, (i, 1, n))
```
Comment (skirpichev) [inline sympy/functions/combinatorial/factorials.py]: Use named Dummy.

**Comment by aktech** [comment]
I think you mean : symbol 'i' (for dummy variable i)

as If I use  `i = Dummy('i')` , the output would look like this:
`Product(_i, (_i, 1, k))`

**Comment by skirpichev** [comment]
> I think you mean : symbol

No.  I mean Dummy.

**Comment by aktech** [comment]
If I use i = Dummy('i') , the output would be `Product(_i, (_i, 1, k))'   (Notice _i)
and if I put that into tests that would give an Error.

**Comment by skirpichev** [comment]
You can compare e.g. strings.  See other similar tests in the function (e.g. Sum rewrite).

**Comment by aktech** [comment]
Oh! fine, that would be good.
Thanks.

**Comment by pelegm** [comment]
+1.

**Comment by aktech** [comment]
Fixed. 
@skirpichev 

Thanks

**Comment by aktech** [comment]
Fixed Merge Conflicts.
@skirpichev

**Pull request merged**